src/Gaufrette/Adapter/AwsS3.php

```diff
@@ -307,22 +309,13 @@ protected function computePath($key)
 
     /**
      * Computes the key from the specified path.
-     *
-     * @param string $path
-     *
-     * return string
      */
-    protected function computeKey($path)
+    protected function computeKey(string $path): string
     {
         return ltrim(substr($path, strlen($this->options['directory'])), '/');
     }
 
-    /**
-     * @param string $content
-     *
-     * @return string
-     */
-    private function guessContentType($content)
+    private function guessContentType(mixed $content): false|string
```

Comment:
not available on 8.0.
```suggestion
    private function guessContentType(mixed $content): bool|string
```